python_scripts/ensemble_hyperparameters.py

```diff
@@ -62,8 +62,15 @@
 
 # %% [markdown]
 # We can observe that in our grid-search, the largest `max_depth` together
-# with the largest `n_estimators` led to the best generalization performance.
-#
+# with the largest `n_estimators` led to the best performance in the validation
+# set. Now we will estimate the generalization performance of the best model
```

Comment:
Since we have cross-validation, then we have several validations sets. So we should use the plural when referring to the set and probably mentioned that the ranking is based on the average.

```
led in average to the best performance on the validation sets
```